### ISSUE TITLE
dispatch_group_enter is not balanced to dispatch_group_leave

### ISSUE THREAD

**Issue by PhilCai1993** [rs/SDWebImage#2247]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

In `UIView+WebCache.m`

**Comment by dreampiggy** [comment]
What version did you use ? And could you please provide the repro rate ? (100% ? Not possible...)

This `SDWebImageInternalSetImageGroupKey ` is used only for `FLAnimatedImageView+WebCache`. And the implementation call `dispatch_group_enter` at the beginning. So it should be balanced.

**Comment by icomico** [comment]
en, i also meet with the same problem, when call this function  
 - (void)sd_internalSetImageWithURL:(nullable NSURL *)url
                  placeholderImage:(nullable UIImage *)placeholder
                           options:(SDWebImageOptions)options
                      operationKey:(nullable NSString *)operationKey
                     setImageBlock:(nullable SDSetImageBlock)setImageBlock
                          progress:(nullable SDWebImageDownloaderProgressBlock)progressBlock
                         completed:(nullable SDExternalCompletionBlock)completedBlock
                           context:(nullable NSDictionary<NSString *, id> *)context;  in  FLAnimatedImageView+WebCache.m,  it crash in dispatch_group_leave(group),   but i didn't call dispatch_group_eneter(group) first;

**Comment by bpoplauschi** [comment]
Yeap, it's a bit weird, but it looks balanced. Not sure what we can do to simplify
